comp/trace/agent/impl/agent.go

```diff
@@ -59,7 +64,27 @@ type dependencies struct {
 	Tagger             tagger.Component
 }
 
-type component struct{}
+type component struct {
+	*agent
+}
```

Comment:
Why do we need this ? Should agent struct is a component ?

Reply:
This is a restriction on fx interfaces. The struct `agent` needs to be stored as a pointer while the struct `component` needs to be pass-by-value.

Reply:
Didn't realize we had component struct. We should remove that. We are returning value rather than pointer, hence the issue (not related to fx). Fx uses the value that is returned from the constructor. 

https://github.com/DataDog/datadog-agent/blob/e9a7db6de2db783d7c34a28169db92d3fac51266/comp/trace/agent/impl/agent.go#L78-L115

Reply:
Found out that we just need to store `pkg.Agent` as a pointer so we can merge the two structs `agent` and `component` here 👍